### PR TITLE
plugins/discovery: ensure discovery doesn't erase its own config

### PR DESCRIPTION
**Pull request by blacksails**
When the discovery plugin receives a a discovery bundle which omits the
discovery configuration, it deletes its own configuration from the
manager. In turn this means that `GET /v1/config` doesn't show the
configuration of the discovery plugin.

This change ensures that the plugin will never overwrite the discovery
configuration on the manager.

Signed-off-by: Benjamin Nørgaard <mail@blacksails.dev>

### CODE REVIEW

**Comment by ashutosh-narkar** [comment]
@blacksails the changes look fine. Can you please fix the broken tests. Thanks.

**Comment by charlieegan3** [comment]
(I think this is all you need)

```patch
diff --git a/sdk/opa_test.go b/sdk/opa_test.go
index 5fbe773c0..63e89bc76 100644
--- a/sdk/opa_test.go
+++ b/sdk/opa_test.go
@@ -161,7 +161,8 @@ func TestHookOnConfigDiscovery(t *testing.T) {
                        "foo":     "baz",
                        "fox":     "quz",
                },
-               Plugins: map[string]json.RawMessage{"test_plugin": json.RawMessage("{}")},
+               Plugins:   map[string]json.RawMessage{"test_plugin": json.RawMessage("{}")},
+               Discovery: json.RawMessage(`{"service":"disco", "resource": "disco.tar.gz"}`),
        }
        act := th1.c // doesn't matter which hook, they only mutate the config via its pointer
        if diff := cmp.Diff(exp, act, cmpopts.IgnoreFields(config.Config{}, "DefaultDecision", "DefaultAuthorizationDecision")); diff != "" {
```

**Comment by blacksails** [comment]
Fixed test and rebased

**Review by charlieegan3 — APPROVED** [review]
This looks good to me 🙂. I'll let Ash have a final check before we merge. Thanks for working on it!

**Review by ashutosh-narkar — APPROVED** [review]
LGTM

**Pull request merged**
